Abyss.Web/ClientApp/src/app/servermanager/servermanager.component.ts

```diff
@@ -7,21 +8,69 @@ import { DialogService } from '../services';
 
 @Component({
     templateUrl: './servermanager.component.html',
+    styleUrls: ['./servermanager.component.scss'],
 })
 export class ServerManagerComponent implements OnInit {
     public servers?: IServer[];
     public ServerStatus = ServerStatus;
     public selectedId?: string;
     public loading = false;
+    public log: string[] = [];
+    public hubReady = false;
+    private hub = new HubConnectionBuilder()
+        .withUrl('hub/servermanager', { transport: HttpTransportType.WebSockets })
+        .withAutomaticReconnect()
+        .build();
 
     public get selected(): IServer | undefined { return this.servers ? this.servers.find(x => x.Id === this.selectedId) : undefined; }
 
-    constructor(public serverManagerService: ServerManagerService, public dialogService: DialogService) { }
+    constructor(public serverManagerService: ServerManagerService, public dialogService: DialogService, private ngZone: NgZone) {
+        this.hub.on('status', (status: ServerStatus) => {
+            this.ngZone.run(() => {
```

Comment:
Why are all of these running outside zone.js?

Reply:
I believe incoming calls from SignalR don't trigger zone so must be invoked manually but maybe that's changed since I last worked on it